models.py

```diff
@@ -28,12 +32,42 @@ def generate_compiled_segmentation_model(model_name, model_parameters, num_class
                                          optimizing_class_id=None, optimizing_input_threshold=None,
                                          optimized_class_thresholds=None):
 
-    # These are the only model, loss, and optimizer currently supported
-    assert model_name == 'Unet'
-    assert loss == 'cross_entropy'
-    assert optimizer == 'adam'
+    # alter input_shape due to inability of yaml to accept tupples!
+    if 'input_shape' in model_parameters:
+        model_parameters['input_shape'] = tuple(model_parameters['input_shape'])
+    else:  # to guarantee the old config files still work
+        model_parameters['input_shape'] = (None, None, 1)
+
+    # These are the only optimizer currently supported
+    if optimizer.lower() == 'adam':
+        opt_fn = Adam()
+    elif optimizer.lower() == 'sdg':
+        opt_fn = SGD()
+    elif optimizer.lower() == 'adagrad':
+        opt_fn = Adagrad()
+    elif optimizer.lower() == 'adamax':
+        opt_fn = Adamax()
+    elif optimizer.lower() == 'ftrl':
+        opt_fn = Ftrl()
+    elif optimizer.lower() == 'nadam':
+        opt_fn = Nadam()
+    elif optimizer.lower() == 'rmsprop':
+        opt_fn = RMSprop()
+    else:
+        raise NameError("Optimizer not supported")
+
+    # These are the only loss currently supported
+    if loss == 'cross_entropy':
+        loss_fn = BinaryCrossentropyL()
+    elif loss == 'cat_cross_entropy':
+        loss_fn == CategoricalCrossentropyL()
```

Comment:
so i previously found in #54 importantly that:

> our last layer is sigmoid, but tf2.1 automatically normalizes the sigmoid into a softmax-like fcn before feeding to cat ce loss fcn

For example, for unet, default activation is sigmoid (which requires binary_ce). If we instead choose categorical ce and sigmoid, tf2.x will convert output to softmax scaling needed for cat_ce.

So, we should assert that if binary_ce is chosen, then so is sigmoid; and if cat_ce is chosen, then so is softmax activation, for clarity to user.

okay @CarolinaFurtado ?